common/lib/dependabot/git_commit_checker.rb

```diff
@@ -365,17 +365,19 @@ def listing_tag_for_version(version)
     def listing_tags
       return [] unless listing_source_url
 
-      tags = listing_repo_git_metadata_fetcher.tags
+      @listing_tags ||= begin
```

Comment:
I believe we've been swapping `||=` memoization to using `if defined?` to handle valid nil values.

Reply:
Hehe, @mattt's cop would've caught me here 😅. I did check that this should always return an array, so I think standard memoization is safe. But I can add a `|| []` if wanted just in case, or a `defined?` check.

Reply:
I think in this case we'll always be dealing with a collection, though that collection may be empty. I'm not sure about our conventions for non-nil value memoization.

Reply:
I think standard memoization should play nice with empty arrays. Here I followed the style in `git_commit_checker.rb` where several collections are memoized in the standard way.

Reply:
Sorry if my comment added confusion. I was responding at the same time as you @deivid-rodriguez and your comment didn't render before I clicked "comment".

I agree with you that we should be fine to use standard memoization (without `|| []`)